packages/store/src/decorators/select/select.ts

```diff
@@ -23,17 +34,25 @@ export function Select<T>(rawSelector?: T, ...paths: string[]): PropertyDecorato
       [name]: {
         enumerable: true,
         configurable: true,
-        get(): PropertyType<T> {
+        get() {
+          if (this[selectorId]) {
+            return this[selectorId];
+          }
           // The `localInject` will be tree-shaken away in apps that
           // still use the View Engine.
-          const store = ɵivyEnabled ? localInject(this, Store) : null;
-          const config = ɵivyEnabled ? localInject(this, NgxsConfig) : null;
-
-          selector = selector || createSelectorFn(config, name, rawSelector, paths);
-
-          return (
-            this[selectorId] || (this[selectorId] = createSelectObservable(selector, store))
-          );
+          else if (ɵivyEnabled) {
+            return (this[selectorId] = injectorNotifier$!.pipe(
+              mergeMap(() => {
+                const store = localInject(this, Store);
+                const config = localInject(this, NgxsConfig);
+                selector = selector || createSelectorFn(config, name, rawSelector, paths);
+                return createSelectObservable(selector, store);
+              })
+            ));
+          } else {
+            selector = selector || createSelectorFn(null, name, rawSelector, paths);
+            return (this[selectorId] = createSelectObservable(selector, null));
+          }
```

Comment:
what do you think?)

```ts
if (!this[selectorId]) {

  if (ɵivyEnabled) {
    this[selectorId] = injectorNotifier$!.pipe(
      mergeMap(() => {
        const store = localInject(this, Store);
        const config = localInject(this, NgxsConfig);
        selector = selector || createSelectorFn(config, name, rawSelector, paths);
        return createSelectObservable(selector, store);
      })
    ));
  } else {
     selector = selector || createSelectorFn(null, name, rawSelector, paths);
     this[selectorId] = createSelectObservable(selector, null)
  }

}

return this[selectorId];
```

Reply:
Yeah, I don't mind changing that. That looks better.

I'll wait for Mark's response and I'll then update the code with your suggestion.